src/tools/cephfs/cephfs-shell

```diff
@@ -773,6 +773,148 @@ class CephFSShell(Cmd):
+                        dest_second_half = file[0].encode('utf-8')
+                        dest_path = dest_first_half + dest_second_half
+                    else:
+                        set_exit_code_msg(msg="failed to create destination"
+                                              " path as more than one"
+                                              " file/dir matched with regex")
+                        return
+                # if it is a dir then take the dir and concatenate it
+                # with the destination path
+                # NOTE: If the dir in both source and destionation are same
+                # then it means dir link should not be inserted into the dir
+                # but the provided destination path should be followed
+                elif is_dir_exists(args.src_path):
+                    dir1 = re.findall(r"[^/]+(?=/$|$)",
+                                      args.src_path.decode('utf-8'))
+                    dir2 = re.findall(r"[^/]+(?=/$|$)",
+                                      args.dest_path.decode('utf-8'))
+                    if dir1[0] == dir2[0]:
+                        dest_path = args.dest_path
+                    else:
+                        dest = args.dest_path.decode('utf-8').\
+                            rstrip('/')
+                        dest_first_half = dest.encode('utf-8') + b'/'
+                        dest_second_half = dir1[0].encode('utf-8')
+                        dest_path = dest_first_half + dest_second_half
+
+            else:
+                # if the destination is not a file or a dir then:
+                # accept it as file so the end part of path cannot have
+                #    a `/` succeeding it.
+                test_path = args.dest_path.decode('utf-8')
+                if test_path[len(test_path) - 1] == '/':
+                    set_exit_code_msg(errno.ENOENT, f"'{test_path}': "
+                                                    f"No such file or "
+                                                    f"directory")
+                    return
+                else:
+                    dest_path = test_path
+        # sym link creating
+        if args.symbolic:
+            if args.force:
+                try:
+                    cephfs.unlink(dest_path)
+                except libcephfs.Error:
+                    pass
+
+            cephfs.symlink(src_path, dest_path)
+
+            if args.verbose:
+                poutput(f"{dest_path.decode('utf-8')} ->"
+                        f" {src_path.decode('utf-8')}")
+        # hard link creation
+        else:
+            if args.force:
+                try:
+                    cephfs.unlink(dest_path)
+                except libcephfs.Error:
+                    pass
+
+            cephfs.link(src_path, dest_path)
+
+            if args.verbose:
+                poutput(f"{dest_path.decode('utf-8')} ->"
+                        f" {src_path.decode('utf-8')}")
+
```

Comment:
This hasn't been covered in the push. We can remove duplication here.

```suggestion
        if args.force:
                try:
                    cephfs.unlink(destination_path)
                     if args.symbolic:
                        cephfs.symlink(source_path, destination_path)
                     else:
                        cephfs.link(source_path, destination_path)
                except libcephfs.Error:
                    # print errmsg and errno here?
                    return

        if args.verbose:
            poutput(f"{destination_path.decode('utf-8')} ->"
                    f" {source_path.decode('utf-8')}")
```

Reply:
Okay